modules/web/src/com/haulmont/cuba/web/gui/components/WebAbstractTable.java

```diff
@@ -2192,9 +2192,10 @@ public boolean saveSettings(Element element) {
         boolean settingsChanged = false;
 
         if (isUsePresentations()) {
-            String textSelection = String.valueOf(component.isTextSelectionEnabled());
-            if (!textSelection.equalsIgnoreCase(element.attributeValue("textSelection"))) {
-                element.addAttribute("textSelection", textSelection);
+            boolean textSelection = component.isTextSelectionEnabled();
+            String settingsTextSelection = element.attributeValue("textSelection");
+            if (textSelection != Boolean.parseBoolean(settingsTextSelection) || settingsTextSelection == null) {
```

Comment:
Since the short-circuit evaluation is applicable to the conditional-OR operation, it'll be more efficient if we check `textSelection == null` first, so we can avoid converting it to the boolean value.

Reply:
fixed